kornia/geometry/transform/pyramid.py

```diff
@@ -410,3 +409,24 @@ def build_laplacian_pyramid(
         laplacian_pyramid.append(laplacian)
     laplacian_pyramid.append(gaussian_pyramid[-1])
     return laplacian_pyramid
+
+
+def upscale_double(x):
+    r"""Upscale image by the factor of 2, even indices maps to original indices.
+
+    Odd indices are linearly interpolated from the even ones.
+
+    Args:
+        x : input image.
+    Shape:
+        - Input: :math:`(B, C, H, W)`
+        - Output :math:`(B, C, H, W)`
+    """
```

Comment:
need to add checkers `KORNIA_CHECK_SHAPE(x, ["*", "H", "W"])`

and i would adjust the function to work with arbitrary 2d shape

Reply:
done